pkg/storage/wal/segment.go

```diff
@@ -49,35 +57,39 @@ type streamSegment struct {
 }
 
 // NewWalSegmentWriter creates a new WalSegmentWriter.
-func NewWalSegmentWriter() *SegmentWriter {
-	return &SegmentWriter{
-		streams: swiss.NewMap[streamID, *streamSegment](64),
-		buf1:    encoding.EncWith(make([]byte, 0, 4)),
+func NewWalSegmentWriter() (*SegmentWriter, error) {
+	idxWriter, err := index.NewWriter(context.TODO())
```

Comment:
Do we have to pass a ctx to the index? I think we can remove it because we only ever pass `context.TODO()`. We also don't pass a context to the parent SegmentWriter so the utility is very limited.

Reply:
Yes I'll look into that.